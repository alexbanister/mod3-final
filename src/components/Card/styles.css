.card {
  background: #999999;
  margin: 1em;
  padding: 1em;
  max-width: 40%;
  border-radius: .5em;
  display: grid;
  grid-template-columns: 1fr 2fr;
  text-align: left;
}
.title, .entry {
  padding: .5em;
}
.title {
  font-weight: bold;
}
.card p {
  margin: 0;
  padding: 0;
}
