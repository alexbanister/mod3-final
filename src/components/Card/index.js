import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { fetchSwornMember } from '../../api';
import './styles.css';

export class Card extends Component {
  constructor(){
    super();
    this.state = {
      swornMembers: []
    };
  }

  showSwornMembers(members) {
    return members.map( (member, index) => {
      return <h5 key={`${member.name}-${index}`}>{member.name}</h5>;
    });
  }

  async getSwornMember(membersURLs) {
    const allMembers = await fetchSwornMember(membersURLs);
    this.setState({
      swornMembers: allMembers
    });
  }

  displayLists = (list) => {
    return list.map( item => {
      return <p key={item}>{item}</p>;
    });
  };

  render() {
    return (
      <div className='card'>
        <div className='title'>
          House:
        </div>
        <div className='entry'>
          {this.props.house.name}
        </div>
        <div className='title'>
          Words:
        </div>
        <div className='entry'>
          {this.props.house.words}
        </div>
        <div className='title'>
          Founded:
        </div>
        <div className='entry'>
          {this.props.house.founded}
        </div>
        <div className='title'>
          Coat Of Arms:
        </div>
        <div className='entry'>
          {this.props.house.coatOfArms}
        </div>
        <div className='title'>
          Seats:
        </div>
        <div className='entry'>
          {this.displayLists(this.props.house.seats)}
        </div>


        <div className='title'>
          Titles:
        </div>
        <div className='entry'>
          {this.displayLists(this.props.house.titles)}
        </div>
        <div className='title'>
          Ancestral Weapons:
        </div>
        <div className='entry'>
          {this.displayLists(this.props.house.ancestralWeapons)}
        </div>
        <div className='swornMembers'>
          <button
            onClick={()=>{ this.getSwornMember(this.props.house.swornMembers); }}>
            Show Sworn Members
          </button>
          {this.showSwornMembers(this.state.swornMembers)}
        </div>
      </div>
    );
  }
}

Card.propTypes = {
  house: PropTypes.object
};

export default Card;
