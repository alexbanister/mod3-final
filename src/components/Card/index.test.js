import { shallow, mount } from 'enzyme';
import Card from './index';
import React from 'react';
import fetchMock from 'fetch-mock';

describe('Card snapshot', () => {

  it('should always match the snapshot', () => {
    const house = {
      name:"House Corbray of Heart's Home",
      coatOfArms:"Three black ravens in flight",
      words:"",
      titles:[
        "Lord of Heart's Home",
        "King of the Fingers (historical)"
      ],
      founded:"",
      seats:[
        "Heart's Home"
      ],
      ancestralWeapons:[
        "Lady Forlorn"
      ]
    };
    const wrapper = shallow(<Card house={house} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    const house = {
      name:"House Corbray of Heart's Home",
      coatOfArms:"Three black ravens in flight",
      words:"",
      titles:[
        "Lord of Heart's Home",
        "King of the Fingers (historical)"
      ],
      founded:"",
      seats:[
        "Heart's Home"
      ],
      ancestralWeapons:[
        "Lady Forlorn"
      ]
    };
    const wrapper = shallow(<Card house={house} />);

    expect(wrapper.instance().props.house).toEqual(house);
  });

  it('should have display sworn members', () => {
    const mockReturn = [{ name: "Corwyn Corbray" }, { name: "Gwayne Corbray" }];
    fetchMock.post('http://localhost:3001/api/v1/character', {
      status: 200,
      body: mockReturn
    });

    const house = {
      name:"House Corbray of Heart's Home",
      coatOfArms:"Three black ravens in flight",
      words:"",
      titles:[
        "Lord of Heart's Home",
        "King of the Fingers (historical)"
      ],
      founded:"",
      seats:[
        "Heart's Home"
      ],
      ancestralWeapons:[
        "Lady Forlorn"
      ],
      swornMembers:[
        "https://www.anapioficeandfire.com/api/characters/1182"
      ]
    };
    const wrapper = shallow(<Card house={house} />);

    expect(wrapper.state()).toEqual({ swornMembers: [] });

    wrapper.find('button').simulate('click');

    console.log(wrapper.debug());
    // expect(wrapper.state()).toEqual({ swornMembers: mockReturn });
  });
});
