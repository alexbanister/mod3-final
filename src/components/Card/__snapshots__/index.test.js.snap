// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    house={
        Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "coatOfArms": "Three black ravens in flight",
            "founded": "",
            "name": "House Corbray of Heart's Home",
            "seats": Array [
              "Heart's Home",
            ],
            "titles": Array [
              "Lord of Heart's Home",
              "King of the Fingers (historical)",
            ],
            "words": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          House Corbray of Heart's Home
</h3>,
        <h3 />,
        <h3 />,
        <h3>
          Three black ravens in flight
</h3>,
        <h3>
          <p>
                    Heart's Home
          </p>
</h3>,
        <h3>
          <p>
                    Lord of Heart's Home
          </p>
          <p>
                    King of the Fingers (historical)
          </p>
</h3>,
        <h3>
          <p>
                    Lady Forlorn
          </p>
</h3>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "House Corbray of Heart's Home",
        },
        "ref": null,
        "rendered": "House Corbray of Heart's Home",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Three black ravens in flight",
        },
        "ref": null,
        "rendered": "Three black ravens in flight",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Heart's Home
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Heart's Home",
            "nodeType": "host",
            "props": Object {
              "children": "Heart's Home",
            },
            "ref": null,
            "rendered": "Heart's Home",
            "type": "p",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Lord of Heart's Home
</p>,
            <p>
              King of the Fingers (historical)
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Lord of Heart's Home",
            "nodeType": "host",
            "props": Object {
              "children": "Lord of Heart's Home",
            },
            "ref": null,
            "rendered": "Lord of Heart's Home",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "King of the Fingers (historical)",
            "nodeType": "host",
            "props": Object {
              "children": "King of the Fingers (historical)",
            },
            "ref": null,
            "rendered": "King of the Fingers (historical)",
            "type": "p",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Lady Forlorn
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Lady Forlorn",
            "nodeType": "host",
            "props": Object {
              "children": "Lady Forlorn",
            },
            "ref": null,
            "rendered": "Lady Forlorn",
            "type": "p",
          },
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            House Corbray of Heart's Home
</h3>,
          <h3 />,
          <h3 />,
          <h3>
            Three black ravens in flight
</h3>,
          <h3>
            <p>
                        Heart's Home
            </p>
</h3>,
          <h3>
            <p>
                        Lord of Heart's Home
            </p>
            <p>
                        King of the Fingers (historical)
            </p>
</h3>,
          <h3>
            <p>
                        Lady Forlorn
            </p>
</h3>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "House Corbray of Heart's Home",
          },
          "ref": null,
          "rendered": "House Corbray of Heart's Home",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Three black ravens in flight",
          },
          "ref": null,
          "rendered": "Three black ravens in flight",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Heart's Home
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Heart's Home",
              "nodeType": "host",
              "props": Object {
                "children": "Heart's Home",
              },
              "ref": null,
              "rendered": "Heart's Home",
              "type": "p",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Lord of Heart's Home
</p>,
              <p>
                King of the Fingers (historical)
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Lord of Heart's Home",
              "nodeType": "host",
              "props": Object {
                "children": "Lord of Heart's Home",
              },
              "ref": null,
              "rendered": "Lord of Heart's Home",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "King of the Fingers (historical)",
              "nodeType": "host",
              "props": Object {
                "children": "King of the Fingers (historical)",
              },
              "ref": null,
              "rendered": "King of the Fingers (historical)",
              "type": "p",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Lady Forlorn
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Lady Forlorn",
              "nodeType": "host",
              "props": Object {
                "children": "Lady Forlorn",
              },
              "ref": null,
              "rendered": "Lady Forlorn",
              "type": "p",
            },
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
